### ISSUE TITLE
Minor API related things found while documenting

### ISSUE THREAD

**Issue by JamesNK** [aspnet/SignalR#2131]
Some minor API things I have noticed while documenting. I don't know if these meet the change bar.

IHubClientsExtensions -> should be named HubClientsExtensions. (since it itself is not an interface)

DynamicHubClients.Users(IReadOnlyList<string> users) -> should be userIds for consistency

**Comment by JamesNK** [comment]
Should `SerializedHubMessage` and `SerializedMessage` located in Core be under the `Microsoft.AspNetCore.SignalR.Protocol` namespace?

**Comment by JamesNK** [comment]
Was there a rule about public types exposing pub-internal types? `ConnectionsRouteBuilder` takes the pub-internal `HttpConnectionDispatcher` in its public ctor

**Comment by JamesNK** [comment]
`RedisDependencyInjectionExtensions.AddRedis` overloads, should rename `builder` to `signalrBuilder` to be consistent with other extension methods.

**Comment by JamesNK** [comment]
`SignalRDependencyInjectionExtensions.AddHubOptions` should rename `options` parameter to `configure` to be consistent with other configuration callbacks.

**Comment by anurse** [comment]
> Not ideal but the ctor is for DI so no change required.

How is it for DI? It's constructed directly in the [one place it's used](https://github.com/aspnet/SignalR/search?utf8=%E2%9C%93&q=ConnectionsRouteBuilder&type=), which is also inside the same assembly. Could the constructor just be internal?

**Comment by JamesNK** [comment]
You're right. I thought `HttpConnectionDispatcher` was passed in via DI but it is resolved manually and then the ctor is used. Will change it to internal.
